editor/components/inserter/menu.js

```diff
@@ -2,20 +2,83 @@
  * External dependencies
  */
 import { connect } from 'react-redux';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/operator/filter';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 import Dashicon from 'components/dashicon';
+import { focusNextFocusableRef, focusPreviousFocusableRef, listVisibleBlocksByCategory } from './helpers';
 
 class InserterMenu extends wp.element.Component {
 	constructor() {
 		super( ...arguments );
+		this.blockTypes = wp.blocks.getBlocks();
+		this.categories = wp.blocks.getCategories();
+		this.nodes = {};
 		this.state = {
-			filterValue: ''
+			filterValue: '',
+			focusedElementRef: null
 		};
 		this.filter = this.filter.bind( this );
+		this.isShownBlock = this.isShownBlock.bind( this );
+		this.changeMenuSelection = this.changeMenuSelection.bind( this );
+		this.setSearchFocus = this.setSearchFocus.bind( this );
+		this.bindReferenceNode = this.bindReferenceNode.bind( this );
+	}
+
+	bindReferenceNode( nodeName ) {
+		const binder = ( node ) => this.nodes[ nodeName ] = node;
+		binder.bind( this );
+
+		return binder;
+	}
+
+	componentDidMount() {
+		// Build a stream of keydown events on component mount.
+		this.keyStream$ = Observable.fromEvent( document, 'keydown' );
```

Comment:
Sounds good.  I can take RxJS out and open a separate issue.  The only problem is we might need to do some event normalization around this for better device compatibility.  Most of the event handlers we are using are either with TinyMCE and React's normalized event system.  This PR is an example where we would need to bind to the `document` to listen for keydowns for this component, and was initially why I thought it would be appropriate to use RxJS.  I will see if I can get React's events to do the same thing, but it might take me a while :).